easy-ui-react/src/Pill/Pill.tsx

```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import CloseIcon from "@easypost/easy-ui-icons/Close";
+import { Icon } from "../Icon";
+import { Text } from "../Text";
+import { IconSymbol } from "../types";
+import { UnstyledButton } from "../UnstyledButton";
+import styles from "./Pill.module.scss";
+
+export type PillProps = {
+  /** Text label */
+  label: string;
+  /** Left aligned icon */
+  icon?: IconSymbol;
+  /** Callback function when dismissing Pill */
+  onDismiss?: () => void;
+};
+
+/**
+ * A `Pill` is a compact element that displays contextual text
+ * typically representing selections or options.
+ *
+ * @remarks
+ * Supports dismissal via `onDismiss` prop and icons via the `icon`
+ * prop.
+ *
+ * @example
+```tsx
+* import { Pill } from "@easypost/easy-ui/Pill";
+*
+* function Component({ carrier }) {
+*   const CarrierImage = () => <Image src={`/${carrier}.png`} />;
+*   return <Pill label="First Last #12345" icon={CarrierImage} />;
+* }
+*/
+export function Pill(props: PillProps) {
```

Comment:
fwiw, i noticed `react-aria` does have a primitive for handling pills/tags with grouping https://react-spectrum.adobe.com/react-aria/useTagGroup.html. for our use case im not entirely sure if its necessary given we're not really handling explicit selection (as far as i know the only _interaction_ we're concerned with is the dismiss button), and there does seem to be a bit of overhead even with just `useTag` as it assumes you are passing along some state. our use case seems to be a bit closer to what `polaris` has https://polaris.shopify.com/components/selection-and-input/tag?example=tag-removable. 

that said, im not opposed to making this a slightly more heavy component with group/list handling because there does seem to be some wins to be had. also we could probably make it a bit flexible by allowing consumers to pass container properties similar to what we do for `HorizontalStack`, `VeritcalStack`..etc, but im also iffy on the way it recommends to handle list state as just feels a bit heavy. idk just thinking out loud lol

Reply:
this is a good point actually. it probably would be good to use React Aria since it does provide some accessibility patterns that would be hard to build ourselves (like keyboard nav). we should maybe consider using `TagGroup` as the name since this should typically be used in a group context

it may be an opportunity to use React Aria Components to make it easier: https://react-spectrum.adobe.com/react-aria/TagGroup.html. with this, i think the work would be mostly limited to styling

Reply:
updated component architecture and used `react-aria-components`